static/js/admin.js

```diff
@@ -889,6 +911,55 @@ function _setup_page() {
         });
     });
 
+    $("#alias_table").on("click", ".delete_alias", function () {
+        var url = "/json/realm/domains/" + $(this).data('id');
+        var aliases_info = $("#realm_aliases_modal").find(".aliases_info");
+
+        channel.del({
+            url: url,
+            success: function () {
+                aliases_info.removeClass("text-error");
+                aliases_info.addClass("text-success");
+                aliases_info.text("Deleted successfully!");
+            },
+            error: function (xhr) {
+                aliases_info.removeClass("text-success");
+                aliases_info.addClass("text-error");
+                aliases_info.text(JSON.parse(xhr.responseText).msg);
+            }
+        });
+    });
+
+    $("#add_alias").click(function () {
+        var aliases_info = $("#realm_aliases_modal").find(".aliases_info");
+        var domain = $("#new_alias").val();
+        var subdomains_allowed = false;
+        if (domain.slice(0, 2) === '*.') {
```

Comment:
hm, not sure I'm a fan of this. For instance, this doesn't make it clear that we do not allow things like `z*p.org`. I would go the route of only allowing valid domains (e.g. enforce the rules at the top of https://www.google.com/search?q=what+characters+are+allowed+in+domains), and then have a checkbox in the table for "allow subdomains".

I'd also just have the row where you add a domain in the table itself, and the Add button can go in the column that the Remove buttons are in now.